enhancements/etcd/assisted-install.md

```diff
@@ -0,0 +1,223 @@
+---
+title: support-assisted-installer
+
+authors:
+  - "@hexfusion"
+reviewers:
+  - "@deads2k"
+  - "@skolicha"
+  - "@ironcladou"
+approvers:
+  - TBD
+creation-date: 2020-09-16
+last-updated: 2020-09-23
+status: provisional
+---
+
+# add cluster-etcd-operator support for assisted installer
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+The assisted installer was designed to utilize three bare-metal machines to
+provision the control-plane. In order to do this after bootstrap is complete
+the bootstrap node is repurposed into the third master node. The assisted
+installer requires etcd to maintain a quorum of two for a short period of
+time until it reprovisions the bootstrap node as the third master.
+
+## Challenges
+The installer currently does not provide an `install_type` key for the
+install-config. So no direct signal is passed from the installer to operator of
+the exact intent to perform a non standard install process.
+
+The result is the operator must make educated assumptions as to the install intent based
+on observations from the cluster itself.
+
+## Motivation
+Currently the `cluster-etcd-operator` does not tolerate less than 3 master
+nodes during install time.
+
+Reasons outlined for cluster with less than 3 master nodes as unsafe non HA.
+1. Violates operational best practices for etcd - unstable
+1. Allows non-HA configurations which we cannot support in production - unsafe and non-HA
+1. Allows a situation where we can get stuck unable to re-achieve quorum, resulting in cluster-death - unsafe, non-HA, non-production, unstable
+1. The combination of all these things makes the situation unsupportable. 
+
+To allow toleration of this check we provided an unsupportedConfigOverrides
+`useUnsupportedUnsafeNonHANonProductionUnstableEtcd`. As assisted installer is
+now moving to GA a supported toleration is required.
+
+## Goals
+
+1.  Provide a supported solution for assisted installer that will allow
+    `cluster-etcd-operator` to tolerate 2 master replicas as install complete.
```

Comment:
The assisted-installer require less than 3 master nodes during the installtion.
By `cluster-etcd-operator` to tolerate 2 master replicas as install complete,
you mean that `/usr/bin/cluster-etcd-operator wait-for-ceo` will return when the cluster has 2 nodes?

Reply:
before etcd scales into the masters on install, we make sure that the node count is as expected[1].

[1] https://github.com/openshift/cluster-etcd-operator/blob/98590e6ecfe282735c4eff01432ae40b29f81202/pkg/etcdenvvar/etcd_env.go#L72